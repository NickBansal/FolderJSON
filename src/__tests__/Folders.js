import "../setupTest";
import React from "react";
import FoldersOrFiles from "../Components/Folders";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<FoldersOrFiles />", () => {
  let wrapper;
  const handleClick = jest.fn();
  const props = {
    filter: "",
    columnReverse: "",
    sortValue: "",
    folderItemElement: "",
    toggleInformation: "",
    handleClick
  };
  beforeEach(() => {
    wrapper = shallow(<FoldersOrFiles {...props} />);
  });

  it("Should create a snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Should sort the data according to name", () => {
    wrapper.setProps({ sortValue: "name" });
    const foldersTitle = wrapper.find(".Folders__name");
    expect(foldersTitle.first().text()).toBe("Cost centres");
    expect(foldersTitle.last().text()).toBe("Public Holiday policy");
  });

  it("Should sort the data according to date added", () => {
    wrapper.setProps({ sortValue: "added" });
    const foldersTitle = wrapper.find(".Folders__name");
    expect(foldersTitle.first().text()).toBe("Expenses");
    expect(foldersTitle.last().text()).toBe("Misc");
  });

  it("Should sort the data according to folder size", () => {
    wrapper.setProps({ sortValue: "size" });
    const foldersTitle = wrapper.find(".Folders__name");
    expect(foldersTitle.first().text()).toBe("Public Holiday policy");
    expect(foldersTitle.last().text()).toBe("Expenses");
  });

  it("Should filter items using the input value", () => {
    wrapper.setProps({ filter: "e" });
    expect(wrapper.find(".Folders__files").length).toBe(3);

    wrapper.setProps({ filter: "em" });
    expect(wrapper.find(".Folders__files").length).toBe(1);

    wrapper.setProps({ filter: "emx" });
    expect(wrapper.find(".Folders__files").length).toBe(0);
  });

  it("Should set the flex-direction style according to the columReverse value", () => {
    wrapper.setProps({ columnReverse: true });
    expect(wrapper.find(".Folders").prop("style")).toHaveProperty(
      "flexDirection",
      "column-reverse"
    );

    wrapper.setProps({ columnReverse: false });
    expect(wrapper.find(".Folders").prop("style")).toHaveProperty(
      "flexDirection",
      "column"
    );
  });

  it("should handle onclick events", () => {
    wrapper
      .find("i")
      .first()
      .simulate("click");
    expect(handleClick).toBeCalledWith(3);

    wrapper
      .find("i")
      .last()
      .simulate("click");
    expect(handleClick).toBeCalledWith(4);
  });

  it("Should change the name of the arrow icon", () => {
    wrapper.setProps({ toggleInformation: true, folderItemElement: 3 });
    expect(
      wrapper
        .find("i")
        .first()
        .prop("className")
    ).toBe("fas fa-chevron-down fa-2x");
    expect(
      wrapper
        .find("i")
        .last()
        .prop("className")
    ).toBe("fas fa-chevron-right fa-2x");

    wrapper.setProps({ toggleInformation: true, folderItemElement: 4 });
    expect(
      wrapper
        .find("i")
        .first()
        .prop("className")
    ).toBe("fas fa-chevron-right fa-2x");
    expect(
      wrapper
        .find("i")
        .last()
        .prop("className")
    ).toBe("fas fa-chevron-down fa-2x");

    wrapper.setProps({ toggleInformation: false, folderItemElement: 4 });
    expect(
      wrapper
        .find("i")
        .first()
        .prop("className")
    ).toBe("fas fa-chevron-right fa-2x");
    expect(
      wrapper
        .find("i")
        .last()
        .prop("className")
    ).toBe("fas fa-chevron-right fa-2x");
  });
});
