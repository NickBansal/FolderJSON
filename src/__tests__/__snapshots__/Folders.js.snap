// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FoldersOrFiles /> Should create a snapshot 1`] = `
<div
  className="Folders"
  style={
    Object {
      "flexDirection": "column",
    }
  }
>
  <div
    className="Folders__files"
    key="Employee Handbook"
  >
    <div
      className="Folders__item"
    >
      <div
        className="Folders__image"
      >
        <img
          alt="pdf"
          src="adobe.png"
        />
        <p>
          412
          mb
        </p>
      </div>
      <div
        className="Folders__information"
      >
        <p
          className="Folders__name"
        >
          Employee Handbook
        </p>
        <p
          className="Folders__added"
        >
          2017-01-06
        </p>
      </div>
    </div>
  </div>
  <div
    className="Folders__files"
    key="Public Holiday policy"
  >
    <div
      className="Folders__item"
    >
      <div
        className="Folders__image"
      >
        <img
          alt="pdf"
          src="adobe.png"
        />
        <p>
          112
          mb
        </p>
      </div>
      <div
        className="Folders__information"
      >
        <p
          className="Folders__name"
        >
          Public Holiday policy
        </p>
        <p
          className="Folders__added"
        >
          2016-12-06
        </p>
      </div>
    </div>
  </div>
  <div
    className="Folders__files"
    key="Expenses"
  >
    <div
      className="Folders__item"
    >
      <div
        className="Folders__image"
      >
        <img
          alt="folder"
          src="folder.png"
        />
        <p>
          900
          mb
        </p>
      </div>
      <div
        className="Folders__information"
      >
        <p
          className="Folders__name"
        >
          Expenses
        </p>
        <p
          className="Folders__added"
        >
          2017-05-02
        </p>
      </div>
      <i
        className="fas fa-chevron-right fa-2x"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="Folders__files"
    key="Cost centres"
  >
    <div
      className="Folders__item"
    >
      <div
        className="Folders__image"
      >
        <img
          alt="csv"
          src="csv.png"
        />
        <p>
          286
          mb
        </p>
      </div>
      <div
        className="Folders__information"
      >
        <p
          className="Folders__name"
        >
          Cost centres
        </p>
        <p
          className="Folders__added"
        >
          2016-08-12
        </p>
      </div>
    </div>
  </div>
  <div
    className="Folders__files"
    key="Misc"
  >
    <div
      className="Folders__item"
    >
      <div
        className="Folders__image"
      >
        <img
          alt="folder"
          src="folder.png"
        />
        <p>
          859
          mb
        </p>
      </div>
      <div
        className="Folders__information"
      >
        <p
          className="Folders__name"
        >
          Misc
        </p>
        <p
          className="Folders__added"
        >
          2015-04-24
        </p>
      </div>
      <i
        className="fas fa-chevron-right fa-2x"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
